runtime/stdlib/test.go

```diff
@@ -0,0 +1,472 @@
+			Value: emulatorBackendExecuteScriptFunction,
+		},
+	}
+
+	return interpreter.NewCompositeValue(
+		inter,
+		getLocationRange,
+		EmulatorBackendType.Location,
+		emulatorBackendTypeName,
+		common.CompositeKindStructure,
+		fields,
+		common.Address{},
+	)
+}
+
+// 'EmulatorBackend.executeScript' function
+
+const emulatorBackendExecuteScriptFunctionName = "executeScript"
+
+const emulatorBackendExecuteScriptFunctionDocString = `execute script function`
+
+var emulatorBackendExecuteScriptFunctionType = func() *sema.FunctionType {
+	// The type of the 'executeScript' function of 'EmulatorBackend' (interface-implementation)
+	// is same as that of 'BlockchainBackend' interface.
+	typ, ok := blockchainBackendInterfaceType.Members.Get(emulatorBackendExecuteScriptFunctionName)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"cannot find type %s.%s",
+			blockchainBackendTypeName,
+			emulatorBackendExecuteScriptFunctionName,
+		))
+	}
+
+	functionType, ok := typ.TypeAnnotation.Type.(*sema.FunctionType)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"invalid type for %s. expected function",
+			emulatorBackendExecuteScriptFunctionName,
+		))
+	}
+
+	return functionType
+}()
+
+var emulatorBackendExecuteScriptFunction = interpreter.NewUnmeteredHostFunctionValue(
+	func(invocation interpreter.Invocation) interpreter.Value {
+		script, ok := invocation.Arguments[0].(*interpreter.StringValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		args, err := arrayValueToSlice(invocation.Arguments[1])
+		if err != nil {
+			panic(errors.NewUnexpectedErrorFromCause(err))
+		}
+
+		var result interpreter.ScriptResult
+
+		testFramework := invocation.Interpreter.TestFramework
+		if testFramework != nil {
+			result = testFramework.RunScript(script.Str, args)
+		} else {
+			panic(interpreter.TestFrameworkNotProvidedError{})
+		}
+
+		err = result.Error
+		if err != nil {
+			// TODO: Revisit this logic
+			if errors.IsUserError(err) {
+				panic(TestFailedError{
+					Err: err,
+				})
+			} else {
+				panic(err)
+			}
+		}
+
+		return createScriptResult(invocation.Interpreter, result.Value)
+	},
+	emulatorBackendExecuteScriptFunctionType,
+)
+
+func arrayValueToSlice(value interpreter.Value) ([]interpreter.Value, error) {
+	array, ok := value.(*interpreter.ArrayValue)
+	if !ok {
+		return nil, errors.NewDefaultUserError("value is not an array")
+	}
+
+	result := make([]interpreter.Value, 0, array.Count())
+
+	array.Iterate(nil, func(element interpreter.Value) (resume bool) {
+		result = append(result, element)
+		return true
+	})
+
+	return result, nil
+}
+
+// createScriptResult Creates a "ScriptResult" using the return value of the executed script.
+//
+func createScriptResult(inter *interpreter.Interpreter, returnValue interpreter.Value) interpreter.Value {
+	// Lookup and get 'ResultStatus.succeeded' value.
+
+	resultStatusConstructorVar := inter.Activations.Find(resultStatusTypeName)
```

Comment:
Maybe just export `Interpreter.findVariable` instead of exporting `Interpreter.activations`

Reply:
Good idea, added with f2ad6e78c3deb87555f9d542e3776c4104072846